### PR TITLE
Search manager refactoring from actual master

### PR DESCRIPTION
**Pull request by medegor44**

### CODE REVIEW

File: src/components/MainContent.jsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      <div className="wrapper alt spotlight style2">
+        <div className="inner">
+          <div className="sectionContentContainer">
+            <ResponsesContainer isFetching={isFetching} responses={responses} />
+          </div>
+        </div>
+      </div>
+      <div className="wrapper spotlight style3">
+        <div className="inner">
+          <div className="sectionContentContainer createPlaylist">
```
Comment (confirmit-horizons) [inline src/components/MainContent.jsx]: createPlaylist classname belongs to CreatePlaylistForm component so make sense to use it inside a component. sectionContentContainer can be still outside

File: src/components/MainContent.jsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import React, { useState } from "react";
+import PropTypes from "prop-types";
+
+import ResponsesContainer from "./search-results/ResponsesContainer";
+import CreatePlaylistForm from "./create-playlist/CreatePlaylistForm";
+import SearchForm from "./search-form/SearchForm";
+
+import "../css/SpotifySongSearch.css";
```
Comment (confirmit-horizons) [inline src/components/MainContent.jsx]: adjust CSS filenames to fit corresponding component names they belong to

File: src/components/search-form/SearchForm.jsx [hunk 3]
```diff
@@ -1,42 +1,78 @@
...
+import parseArtistsTracks from "../../utils/parser";
+import { fetchTracks } from "../../spotify-client";
+import UnauthorizedError from "../../errors/UnauthorizedError";
 import useToken from "../../hooks/useToken";
 import UserContext from "../../contexts/UserContext";
 
-const SearchForm = ({ handleTracksChange, rawText, handleClick }) => {
+const SearchForm = ({ disabled, setIsFetching, setResponses }) => {
+  const [rawText, setRawText] = useState("");
   const [token] = useToken();
```
Comment (confirmit-horizons) [inline src/components/search-form/SearchForm.jsx:17]: nit: you can move useToken inside performRequest as it's just a hook and it's not used anywhere else

**Review by confirmit-horizons — CHANGES_REQUESTED** [review]

File: src/components/MainContent.jsx (hunk 1 above)
Comment (medegor44) [reply src/components/MainContent.jsx]: I just removed `createPlaylist` style.

File: src/components/search-form/SearchForm.jsx (hunk 3 above)
Comment (medegor44) [reply src/components/search-form/SearchForm.jsx:17]: But performRequest is not a hook, it is just a callback and I can't call hook from function wich is not component.

**Review by confirmit-horizons — APPROVED** [review]

**Pull request merged**